src/main/java/com/google/googleidentity/oauth2/endpoint/AuthorizationEndpoint.java

```diff
@@ -0,0 +1,91 @@
+/*
+    Copyright 2020 Google LLC
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+package com.google.googleidentity.oauth2.endpoint;
+
+import com.google.googleidentity.oauth2.client.ClientDetailsService;
+import com.google.googleidentity.oauth2.exception.OAuth2ExceptionHandler;
+import com.google.googleidentity.oauth2.exception.OAuth2Exception;
+import com.google.googleidentity.oauth2.util.OAuth2ParameterNames;
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import org.apache.http.HttpStatus;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.logging.Logger;
+
+import static com.google.googleidentity.oauth2.exception.OAuth2ExceptionHandler.ErrorCode;
+
+/**
+ * Demo AuthorizationEndpoint for OAuth2 Server
+ */
+@Singleton
+public final class AuthorizationEndpoint extends HttpServlet {
+
+    private static final long serialVersionUID = 5L;
+
+    private static final Logger log = Logger.getLogger("AuthorizationEndpoint");
+
+    private final ClientDetailsService clientDetailsService;
+
+
+    @Inject
+    public AuthorizationEndpoint(ClientDetailsService clientDetailsService) {
+        this.clientDetailsService = clientDetailsService;
+    }
+
+
+    protected void doGet(HttpServletRequest request, HttpServletResponse response)
+            throws ServletException, IOException, UnsupportedOperationException{
+
+        try {
+            String responseType = request.getParameter(OAuth2ParameterNames.RESPONSE_TYPE);
+
+            if (responseType == null) {
```

Comment:
Instead of do inline checks, I think it's better to create a validator for each endpoint to validate the parameters. These validator shall only check if passed in parameters are provided as correct combinations, for example, one endpoint accepts (paramA + paramB) or (paramB + paramD and shall not have paramA) then this validator will ensure the passed in parameters meet these combinations. 

The validator doesn't need to validate the value of each param though, like it doesn't validate if client id and client secret are valid, it just need to ensure these params are set. Though it may do some sanitizing check, for example, if we know all our client ids are start with a certain prefix, then the validator can do such check here.